### ISSUE TITLE
Notify free users in some way when they're nearing their bandwidth cap

### ISSUE THREAD

**Issue by myleshorton** [getlantern/lantern#4309]
A user is using Lantern free and is getting close to their bandwidth cap. In order to avoid hitting the cap, they either want to curtail their usage or upgrade to pro. In both cases, they simply need to know they're getting close to hitting the cap in some fashion.

**Comment by myleshorton** [comment]
Let's handle this on the http proxies themselves.

**Comment by myleshorton** [comment]
We first need some design for this @Derekf5 and @carsonefv

**Comment by oxtoacart** [comment]
Setting to 1 story point for design, will reestimate once design is available.

**Comment by myleshorton** [comment]
I can take this one whenever you're done @Derekf5 and @carsonefv!

**Comment by Derekf5** [comment]
sounds good @myleshorton,  Ideally it sounds like we would have a reaching the limit warning notification, as well as some change once that limit is hit...

**Comment by oxtoacart** [comment]
If we don't limit bandwidth in the first version of this, my proposal may not work, but for a final version it might be nice to have some sort of visual gauge with numeric feedback on remaining bandwidth.  Once someone pays for pro, this could morph into our gauge for remaining time or something like that, which both provides continuity and a nice reminder of the difference between free and pro.

**Comment by Derekf5** [comment]
Thanks @oxtoacart , yes planning on some sort of linear counter...
@myleshorton is this going to be a monthly cap? will it top up at the beginning of each month?

**Comment by myleshorton** [comment]
Yeah it'll be a monthly cap @Derekf5

**Comment by Derekf5** [comment]
Hey @myleshorton Here is a solution, I like this version because it is simple, but informative.  In addition to this we could serve snackbar notification when the user gets close to the limit.  If you like this, i can write some quick specs for you.
<img width="1417" alt="screen shot 2016-05-17 at 12 42 39 pm" src="https://cloud.githubusercontent.com/assets/7864729/15330932/dfe1392a-1c2c-11e6-9f52-841eb0020ff5.png">

**Comment by oxtoacart** [comment]
@Derekf5 I like it!  It hints at a bit of a retro hi-tech aesthetic, reminding me a little bit of the kind of control panels on display in Shadowrun but without going too far:

![image](https://cloud.githubusercontent.com/assets/5000654/15331196/cdc1ed7c-1c25-11e6-9fd4-21b9c92aa39f.png)

**Comment by Derekf5** [comment]
Haha yeah @oxtoacart! Just deleted that version because i saw some quick tweaks i need to make, more soon.

**Comment by Derekf5** [comment]
OK @oxtoacart @myleshorton here is the tweaked version, I killed the icon because i thought it was redundant, Made the progress bar thinner, and rearranged the text.   Let me know if this looks ready for specs to you.
<img width="1391" alt="screen shot 2016-05-17 at 2 11 45 pm" src="https://cloud.githubusercontent.com/assets/7864729/15333893/8e06381e-1c39-11e6-9c23-6c4d9f9814b2.png">

**Comment by oxtoacart** [comment]
I still like it.  Do you want the color of the bar to change when user gets close to cap?

**Comment by Derekf5** [comment]
Yes i was thinking we would change it from the yellow used in the ui to the pink once the user is within 50mb of going over.

**Comment by Derekf5** [comment]
This is all spec'd in the "Pro Desktop (without everfeed) updates 05-17-16" google doc.

**Comment by myleshorton** [comment]
Awesome @Derekf5 thanks!

**Comment by xiam** [comment]
Already merged here https://github.com/getlantern/lantern-desktop-ui/pull/62

**Comment by sunshine19090** [comment]
@xiam @Derekf5 The number is now changing depending on the activity. But the color of the bar is always black and not yellow. Shouldn't it be yellow when the user has 500 MB remaining? And the yellow bar decreases to show the remaining MB? 

Also, are we including data cap for the unblocked sites too?

**Comment by sunshine19090** [comment]
Ignore my previous comment. I see that the yellow line purpose is to show how much the user has used. Shall I close it @Derekf5 ?

**Comment by Derekf5** [comment]
sure @sunshine19090 !
